pyomo/solvers/plugins/solvers/direct_or_persistent_solver.py

```diff
@@ -120,6 +120,7 @@ def _presolve(self, **kwds):
         warmstart_flag = kwds.pop('warmstart', False)
         self._keepfiles = kwds.pop('keepfiles', False)
         self._save_results = kwds.pop('save_results', True)
+        _integer_only_warmstarts = kwds.pop('integer_only_warmstarts', False)
```

Comment:
I think we should follow the same pattern of setting `self. _integer_only_warmstarts` instead. That way, we aren't changing the `_warm_start()` API for specific solvers.

Doing so is resulting in a broken pipeline:

```python
TypeError: _warm_start() got an unexpected keyword argument 'integer_only_warmstarts'
```